### DISCUSSION TITLE
How does frac_veg_nosno and exposed vegetation work in CTSM?

### DISCUSSION THREAD

**Discussion by rgknox** [ESCOMP/CTSM#2696] [General]
I'm going through the code, trying to understand canopystate_inst%frac_veg_nosno_patch(p), how its used, where it is set, and how it interacts with other variables.  Here are my notes:

-------------------------
Key Points:

1) canopystate_inst%frac_veg_nosno_patch sets the exposed vegetation filer, the call to which is here in clm_driver():  https://github.com/ESCOMP/ctsm/blob/master/src/main/clm_driver.F90#L434

>       call setExposedvegpFilter(bounds_clump, &
>           canopystate_inst%frac_veg_nosno_patch(bounds_clump%begp:bounds_clump%endp))

2) canopystate_inst%frac_veg_nosno_patch is set directly from the sister-variable, canopystate_inst%frac_veg_nosno_alb_patch(p) and patch%active:

https://github.com/ESCOMP/ctsm/blob/master/src/main/clm_driver.F90#L1379

>      do p = bounds%begp,bounds%endp
>         if (patch%active(p)) then
>            frac_veg_nosno(p) = frac_veg_nosno_alb(p)
>         else
>            frac_veg_nosno(p) = 0._r8
>         end if
>      end do

-------------------------
Where it is defined:

https://github.com/ESCOMP/ctsm/blob/master/src/biogeophys/CanopyStateType.F90#L25

     integer  , pointer :: frac_veg_nosno_patch     (:)   ! patch fraction of vegetation not covered by snow (0 OR 1) [-] 
     integer  , pointer :: frac_veg_nosno_alb_patch (:)   ! patch fraction of vegetation not covered by snow (0 OR 1) [-] 

--------------------------

Where it is set:

As mentioned above, frac_veg_nosno_patch is a result of frac_veg_nosno_alb_patch, and patch%active, set early in the main driver loop.

frac_veg_nosno_alb_patch is set depending on namelist settings, but is either calculated by:
1) SatellitePhenology
2) CNVEGStructUpdate or
3) clmfates_interface.

-----------------------------

Where it is used (including the resulting filter):

It does not seem like frac_veg_nosno_alb_patch is used anywhere.  I'm guessing we need frac_veg_nosno_alb_patch because it may be filled from the satellite dataset, and may not know at the time of filling it, if patches are active or not?

frac_veg_nosno_patch is used to filter patches that should both have an active land-model, and where snow is not covering all of the vegetation.

--------------------------------
Thoughts?

Maybe expand the in code comments to make it clear that frac_veg_nosno_patch is the key variable that sets the exposed vegetation filter, and that frac_veg_nosno_alb_patch and patch%active are the variables that define it.

**Comment by ekluzek** [comment]
Moving to a discussion.
